fhttp/http_utils.go

```diff
@@ -232,48 +232,60 @@ func parseEntry(entry, about, input string, parseFunc func(input string) (int64,
 // entry parsed by parseFunc
 // when parseFromFormattedString raise error, return -1, error(not nil)
 func parseFormattedString(input, about string, parseFunc func(string) (int64, error)) (int64, error) {
+	var err error
+	var lst []string
+	var weights []float32
+	var values []int64
+	var lastPercent float64
+	var res float32
 	// Input is empty case:
 	if len(input) == 0 {
-		return -1, errors.New("input is empty")
+		err = errors.New("input is empty")
+		goto ParseError
 	}
-	lst := strings.Split(input, ",")
+	lst = strings.Split(input, ",")
 	log.Debugf("Parsing %s %s -> %v", about, input, lst)
 	// Simple non probabilistic status case:
 	if len(lst) == 1 && !strings.ContainsRune(input, ':') {
-		s, err := parseFunc(input)
-		if err != nil {
-			s = -1
+		s, e := parseFunc(input)
+		if e != nil {
+			err = e
+			goto ParseError
 		}
-		return s, err
+		return s, nil
 	}
 	// Parse each entry
-	weights := make([]float32, len(lst))
-	values := make([]int64, len(lst))
-	lastPercent := float64(0)
-	i := 0
-	for _, entry := range lst {
-		s, p, err := parseEntry(entry, about, input, parseFunc)
-		if err != nil {
-			return -1, err
+	weights = make([]float32, len(lst))
+	values = make([]int64, len(lst))
+	lastPercent = float64(0)
+	for i, entry := range lst {
+		s, p, e := parseEntry(entry, about, input, parseFunc)
+		if e != nil {
+			err = e
+			goto ParseError
```

Comment:
I'm not a big fan of goto but maybe in this case it's ok (usually return/break is enough- sometime adding an intermediate function helps too)

Reply:
@ldemailly Thanks for the feedback !!
I rewrote to

- refactoring how to return err
- adding an intermediate function

Please let me know if you have any concerns.